src/main/java/com/amazon/opendistroforelasticsearch/security/compliance/ComplianceConfig.java

```diff
@@ -206,6 +204,36 @@ public static ComplianceConfig from(Settings settings) {
                 index);
     }
 
+    /**
+     * Create compliance configuration from auditModel
+     * saltAsString - Read from settings. Not hot-reloaded. Used for anonymization of fields in FLS using consistent hash.
+     * opendistrosecurityIndex - used to determine if internal index is written to or read from.
+     * type - checks if log destination used is internal elasticsearch.
+     * index - the index used for storing audit logs to avoid monitoring it.
+     * @param auditModel audit model
+     * @param settings settings
+     * @return ComplianceConfig
+     */
+    public static ComplianceConfig from(AuditModel auditModel, Settings settings) {
+        final String opendistrosecurityIndex = settings.get(ConfigConstants.OPENDISTRO_SECURITY_CONFIG_INDEX_NAME, ConfigConstants.OPENDISTRO_SECURITY_DEFAULT_CONFIG_INDEX);
+        final String type = settings.get(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_TYPE_DEFAULT, null);
+        final String index = settings.get(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DEFAULT_PREFIX + ConfigConstants.OPENDISTRO_SECURITY_AUDIT_ES_INDEX, "'security-auditlog-'YYYY.MM.dd");
+
```

Comment:
These 3 params 
- opendistrosecurityIndex
- type
- index
 are fed from elasticsearch.yml. 
They are not reloadable but they are needed to determine if internal index is being updated or the current index is storing audit logs if destination is set to internal elasticsearch (ie: dont audit log an index used for storing audit logs) .